### PR TITLE
Revert "poetry 1.2.0"

### PR DESCRIPTION
**Pull request by ghyde**
This pull request reverts poetry from `1.2.0` to `1.1.15`. `1.2.0` currently breaks bash completion for poetry, resulting in the error:

```
/usr/local/etc/bash_completion.d/poetry: line 40: syntax error near unexpected token `clear'
/usr/local/etc/bash_completion.d/poetry: line 40: `            (cache clear)'
```

This can be reproduced by running the command `bash /usr/local/etc/bash_completion.d/poetry`.

### Package Versions

Here are the versions of poetry and bash used to generate the above error.

#### Poetry

```
$ poetry --version
Poetry (version 1.2.0)

$ brew info poetry
==> poetry: stable 1.2.0 (bottled), HEAD
Python package management tool
https://python-poetry.org/
/usr/local/Cellar/poetry/1.2.0 (2,560 files, 31.9MB) *
  Poured from bottle on 2022-09-04 at 22:55:07
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/poetry.rb
License: MIT
==> Dependencies
Required: python@3.10 ✔, six ✔
==> Options
--HEAD
	Install HEAD version
==> Caveats
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
==> Analytics
install: 12,806 (30 days), 30,131 (90 days), 89,466 (365 days)
install-on-request: 12,728 (30 days), 29,938 (90 days), 88,872 (365 days)
build-error: 11 (30 days)
```

#### Bash

```
$ bash --version
GNU bash, version 5.1.16(1)-release (x86_64-apple-darwin19.6.0)
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

$ brew info bash
==> bash: stable 5.1.16 (bottled), HEAD
Bourne-Again SHell, a UNIX command interpreter
https://www.gnu.org/software/bash/
/usr/local/Cellar/bash/5.1.16 (157 files, 10.9MB) *
  Poured from bottle on 2022-01-06 at 09:39:24
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/bash.rb
License: GPL-3.0-or-later
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 23,257 (30 days), 68,657 (90 days), 319,532 (365 days)
install-on-request: 19,425 (30 days), 56,614 (90 days), 262,508 (365 days)
build-error: 39 (30 days)
```

### CODE REVIEW

**Review by carlocab — CHANGES_REQUESTED** [review]
I don't think we want to revert this. CI also won't pass anyway.

This should be reported upstream and fixed there. We can backport any necessary patches when that's done.

**Comment by cho-m** [comment]
Upstream `poetry` issue https://github.com/python-poetry/poetry/issues/4572 was determined to be a `cleo` issue and fixed in https://github.com/python-poetry/cleo/pull/247

Can try backporting patch for `cleo` resource.

**Comment by Secrus** [comment]
Hi, Cleo and Poetry maintainer here. We are closing in on releasing Cleo 1.0 stable version and it will be backported to Poetry 1.2 as soon as it is available. You can track the progress of getting Cleo to 1.0 here: https://github.com/python-poetry/cleo/issues/248

**Comment by iMichka** [comment]
Closing this PR as we are now waiting for a new release with the fixes or waiting for a patch that can be applied to the current version.

**Pull request closed without merging**
